dask/array/core.py

```diff
@@ -1336,7 +1374,7 @@ def _blocks(self, index):
         layer = {(name,) + key: tuple(new_keys[key].tolist()) for key in keys}
 
         graph = HighLevelGraph.from_collections(name, layer, dependencies=[self])
-        return Array(graph, name, chunks, self.dtype)
+        return Array(graph, name, chunks, meta=self._meta)
```

Comment:
We might include the following lines in the meta normalization code:

```python
if hasattr(meta, '_meta'):
    meta = meta._meta
```

And then use `meta=self` in cases like this.  I think that Dask DataFrame does this today.

Reply:
Will try to do it, there's a few more cleanings I need to do anyway. Thanks for the suggestion!